pkg/migrations/dbactions.go

```diff
@@ -840,3 +840,77 @@ func (a *setReplicaIdentityAction) Execute(ctx context.Context) error {
 		identitySQL))
 	return err
 }
+
+type alterReferencesAction struct {
+	conn   db.DB
+	table  string
+	column string
+}
+
+func NewAlterReferencesAction(conn db.DB, table, column string) *alterReferencesAction {
+	return &alterReferencesAction{
+		conn:   conn,
+		table:  table,
+		column: column,
+	}
+}
+
+func (a *alterReferencesAction) Execute(ctx context.Context) error {
+	definitions, err := a.constraintDefinitions(ctx)
+	if err != nil {
+		return err
+	}
+	for _, def := range definitions {
+		// Drop the existing constraint
+		_, err := a.conn.ExecContext(ctx, fmt.Sprintf("ALTER TABLE %s DROP CONSTRAINT %s",
+			pq.QuoteIdentifier(def.table),
+			pq.QuoteIdentifier(def.name),
+		))
+		if err != nil {
+			return fmt.Errorf("dropping constraint %s on %s: %w", def.name, def.table, err)
+		}
+
+		// Recreate the constraint with the table and new column
+		newDef := strings.ReplaceAll(def.def, a.column, pq.QuoteIdentifier(TemporaryName(a.column)))
+		newDef = strings.ReplaceAll(newDef, a.table, pq.QuoteIdentifier(a.table))
+		_, err = a.conn.ExecContext(ctx, fmt.Sprintf("ALTER TABLE %s ADD CONSTRAINT %s %s",
+			pq.QuoteIdentifier(def.table),
+			pq.QuoteIdentifier(def.name),
+			newDef,
+		))
+		if err != nil {
+			return fmt.Errorf("altering references for %s.%s: %w", a.table, a.column, err)
+		}
+	}
+	return nil
+}
+
+type constraintDefinition struct {
+	name  string
+	table string
+	def   string
+}
+
+func (a *alterReferencesAction) constraintDefinitions(ctx context.Context) ([]constraintDefinition, error) {
+	rows, err := a.conn.QueryContext(ctx, fmt.Sprintf(`
+SELECT conname, r.conrelid::regclass, pg_catalog.pg_get_constraintdef(r.oid, true) as condef
+FROM pg_catalog.pg_constraint r
+WHERE confrelid = %s::regclass AND r.contype = 'f'`,
+		pq.QuoteIdentifier(a.table),
```

Comment:
If we need this information, I think it should be present in the schema returned by `read_schema`. If we are missing information there we should add it to `read_schema` rather than running ad-hoc queries during the migration.